ring_test.go

```diff
@@ -113,6 +113,45 @@ var _ = Describe("Redis Ring", func() {
 		Expect(ringShard2.Info(ctx, "keyspace").Val()).To(ContainSubstring("keys=100"))
 	})
 
+	Describe("[new] dynamic setting ring shards", func() {
+		It("uses one shard after setting it to one shard", func() {
+			Expect(ring.Len(), 2)
+			ring.SetAddrs(map[string]string{
+				"ringShardOne": ":" + ringShard1Port,
+			})
+			Expect(ring.Len(), 1)
```

Comment:
Maybe ring should export shards getter for tests and test should verify name+address as well as shard reuse...

Reply:
@vmihailenco wdyt?
It's an API change.

Reply:
> Maybe ring should export shards getter for tests
> It's an API change.

This could be done without API change using `export_test.go` trick https://medium.com/@robiplus/golang-trick-export-for-test-aa16cbd7b8cd